pkg/cloud/services/wait/wait_test.go

```diff
@@ -14,6 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// nolint:goerr113
```

Comment:
Why skip the linting here?

Reply:
It seems like this could be changed to leverage errors.Is and errors.As

Reply:
It's `nonRetryableError := errors.New("non retryable error")` that is being complained about. It seemed odd to move it to outside the test function.

Reply:
I think it's likely fine to make them package level variables, but I would suggest that we should likely change this to a test-specific package, which should be fine since we are only testing a public method here.

Reply:
Fixed, I think :)